### ISSUE TITLE
Throw `TypeError` when using `redirect` in Server Action

### ISSUE THREAD

**Issue by mogeko** [mogeko/bookworm#90]
when we using `redirect` in Server Action:

https://github.com/mogeko/bookworm/blob/ecb68ff2cb452785890f34ce25d366e11d47770e/apps/web/app/actions.ts#L10-L38

https://github.com/mogeko/bookworm/blob/ecb68ff2cb452785890f34ce25d366e11d47770e/apps/web/app/actions.ts#L69-L76

It will throw a `TypeError`:

```
failed to get redirect response TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11576:11) {
  cause: RequestContentLengthMismatchError: Request body length does not match content-length header
      at write (node:internal/deps/undici/undici:10059:41)
      at _resume (node:internal/deps/undici/undici:10037:33)
      at resume (node:internal/deps/undici/undici:9938:7)
      at connect (node:internal/deps/undici/undici:9927:7) {
    code: 'UND_ERR_REQ_CONTENT_LENGTH_MISMATCH'
  }
}
```

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:19 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T8103
    Binaries:
      Node: 20.5.0
      npm: 9.8.0
      Yarn: 1.22.19
      pnpm: 8.6.11
    Relevant Packages:
      next: 13.4.12
      eslint-config-next: 13.4.12
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A


warn  - Latest canary version not detected, detected: "13.4.12", newest: "13.4.13-canary.9".
        Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
        Read more - https://nextjs.org/docs/messages/opening-an-issue
```

**Comment by mogeko** [comment]
Then I tried to bump the version of Next from `13.4.12` to `13.4.13-canary.9`.

```bash
pnpm --filter @mogeko/bookworm-web install next@13.4.13-canary.9
```

Experiment again and found that the bug has been resolved.

**Comment by mogeko** [comment]
This bug should be resolved by this PR: https://github.com/vercel/next.js/pull/49483

It has been merged into the Canary branch. It should be released as stable version 13.4.3 in the future.
